### DISCUSSION TITLE
Dictionary type errors

### DISCUSSION THREAD

**Discussion by The1029** [retextjs/retext#96] [Q&A] [answered]
Just wanted to check I'm not missing something obvious - first time using retext ecosystem.

I'm trying to use retext-spell, but despite importing a dictionary as described in the readme retext-spell is throwing an error - looking through retext-spell's code, it's expecting typeof dictionary to be a function, but the dictionary modules at https://github.com/wooorm/dictionaries seem to export objects.

My code is exactly as per the retext-spell readme, except that I am using unified (have also tried with retext which gives the same issue).

**Comment by ChristianMurphy** [comment]
Welcome @The1029! 👋 
Sorry you ran into some confusion.

Could you share the code you are trying to run, the error message you are seeing, and what runtime (Node, Deno, Chrome, etc) you are using?

**Reply by The1029** [reply]
Thanks for getting back to me. This is running in Node.

```js
// index.js
import { unified } from 'unified';
import retextEnglish from 'retext-english';
import retextSpell from 'retext-spell';
import retextStringify from 'retext-stringify';
import enGb from 'dictionary-en-gb';

const message = unified()
    .use(retextEnglish)
    .use(retextSpell, { dictionary: enGb })
    .use(retextStringify)
    .process('My very easy methud just shows us nine plannets');

console.log(message);
```

```bash
# CL output
PS ...\retext-cli-test> node index.js
file:///.../node_modules/retext-spell/lib/index.js:106
    throw new TypeError('Missing `dictionary` in options')
          ^

TypeError: Missing `dictionary` in options
    at Function.retextSpell (.../node_modules/retext-spell/lib/index.js:106:11)
    at Function.freeze (.../node_modules/unified/lib/index.js:636:36)
    at Function.process (.../node_modules/unified/lib/index.js:716:10)
    at file:///.../retext-cli-test/index.js:9:6

Node.js v20.10.0
```

```json
// package.json
{
  "name": "retext-cli-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "dictionary-en-gb": "^3.0.0",
    "retext-english": "^5.0.0",
    "retext-spell": "^6.0.0",
    "retext-stringify": "^4.0.0",
    "unified": "^11.0.4"
  }
}

```

**Reply by ChristianMurphy** [reply, marked as answer]
Thanks @The1029!
A couple issues
1. there seems to be a mismatch between the dictionary type in retext-spell and dictionaries.
TypeScript highlighted this, but JS fails as you described.
2. `unified().process()` is a promise and needs to be awaited. 

To use the versions of `dictionary-en-gb` and `retext-spell` from you package json, use the following snippet.

```ts
import { unified } from 'unified';
import retextEnglish from 'retext-english';
import retextSpell from 'retext-spell';
import retextStringify from 'retext-stringify';
import enGb from 'dictionary-en-gb';

const message = await unified()
  .use(retextEnglish)
  .use(retextSpell, { dictionary: (onload) => onload(null, enGb) })
  .use(retextStringify)
  .process('My very easy methud just shows us nine plannets');

console.log(JSON.stringify(message));
```

@wooorm is this an issue where there is a release planned but not yet cut for one of retext-spell or dictionaries? Or is this something that should be updated in the documentation?

**Reply by The1029** [reply]
Thanks for the swift reply, Christian. This works.
